model_analyzer/monitor/__init__.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numba.cuda
+
+
+class Monitor:
```

Comment:
> Why does this go in in the `__init__.py` file? I would expect to be in a `model.py` file.

We learned about modern organization and packaging practices for python as part of the tritonclient packaging. @tanmayv25  can share more of why __init__.py is used and in general how to organize.

Reply:
I would love see what @tanmayv25 has found out.

Reply:
There is no hard and fast rule for writing `__init__.py`  for packages and submodule. Depending upon the nature of the project we can decide upon the following alternatives:
1. Leaving the an empty `__init__.py`: https://github.com/python/cpython/blob/master/Lib/urllib/__init__.py
2. Specifying import orders in `__init__.py`: https://github.com/axialmarket/fsq/blob/master/fsq/__init__.py
3. Include core API implementation in the `__init__.py` : https://github.com/python/cpython/tree/master/Lib/collections

All of the alternatives are widely used as demonstrated by the attached popular examples. I chose chose the third option for tritonclient for following reasons:
1. Maintaining the backwards-compatibility with our earlier package distribution where there was a separate package for each grpc,  http, utils and shmutils.
2. Keeping the core API of grpc and utils, to be separate from additional modules like service_pb2, service_pb2_grpc, shared_memory and cuda_shared_memory. The more advanced user can look into these modules to figure out these features. Next the choice of having `http/__init__.py` instead of http.py came up to stay consistent. 

In summary, the standard doesn't have a hard requirement to keep implementation in a certain order. Quite in contrast with the zen of python `There should be one-- and preferably only one --obvious way to do it`. It entirely depends upon the requirement of the project. Our client code is relatively simple and hence I didn't see any harm in this implementation.  For complex project with lots of nested modules, the third alternative doesn't fair up well as the `__init__.py` of all the parent modules will be called upon an import of a child.

Reply:
@tanmayv25 Thank you very much for the detailed comments. I think in our case I can move it to a `model.py` file and use the second alternative mentioned.